core/tx_list.go

```diff
@@ -279,15 +279,21 @@ func (l *txList) Add(tx *types.Transaction, priceBump uint64) (bool, *types.Tran
 	// If there's an older better transaction, abort
 	old := l.txs.Get(tx.Nonce())
 	if old != nil {
-		// threshold = oldGP * (100 + priceBump) / 100
+		// thresholdFeeCap = oldFC  * (100 + priceBump) / 100
 		a := big.NewInt(100 + int64(priceBump))
-		a = a.Mul(a, old.GasPrice())
+		aFeeCap := new(big.Int).Mul(a, old.FeeCap())
+		aTip := a.Mul(a, old.Tip())
+
+		// thresholdTip    = oldTip * (100 + priceBump) / 100
 		b := big.NewInt(100)
-		threshold := a.Div(a, b)
-		// Have to ensure that the new gas price is higher than the old gas
-		// price as well as checking the percentage threshold to ensure that
+		thresholdFeeCap := aFeeCap.Div(aFeeCap, b)
+		thresholdTip := aTip.Div(aTip, b)
+
+		// Have to ensure that either the new fee cap or tip is higher than the
+		// old ones as well as checking the percentage threshold to ensure that
 		// this is accurate for low (Wei-level) gas price replacements
-		if old.GasPriceCmp(tx) >= 0 || tx.GasPriceIntCmp(threshold) < 0 {
+		if (old.FeeCapCmp(tx) >= 0 || tx.FeeCapIntCmp(thresholdFeeCap) < 0) ||
```

Comment:
It can be problematic if both `FeeCap` and `Tip` are less than the old one. The check is skipped.

Reply:
IMO we could break out the 'trivial' checks to happen first, making this if statement smaller and avoiding arithmetics. 

```golang
	if old != nil {
		if old.FeeCapCmp(tx) >= 0 || old.TipCmp(tx) >= 0 { 
			return false, nil
		} 
		// thresholdFeeCap = oldFC * (100 + priceBump) / 100
		...
		if tx.FeeCapIntCmp(thresholdFeeCap) < 0) || tx.TipIntCmp(thresholdTip) < 0){
```